compiler/noirc_evaluator/src/ssa/ir/printer.rs

```diff
@@ -146,7 +146,9 @@ pub(crate) fn display_instruction(
             writeln!(f, "truncate {value} to {bit_size} bits, max_bit_size: {max_bit_size}",)
         }
         Instruction::Constrain(lhs, rhs, message) => match message {
-            Some(message) => writeln!(f, "constrain {} == {} '{message}'", show(*lhs), show(*rhs)),
+            Some(message) => {
+                writeln!(f, "constrain {} == {} '{message:?}'", show(*lhs), show(*rhs))
```

Comment:
Seems like we should format the message similarly to an `Instruction::Call`

Reply:
This will make the SSA for each constrain instruction with a dynamic assertion possibly quite long as the call inputs also contain the type metadata for resolution of strings. 

For example here is the SSA for `assert_msg_runtime` under compile_failure:
```
After Dead Instruction Elimination:
acir fn main f0 {
  b0(v0: Field, v1: Field):
    v108 = eq v0, v1
    v109 = not v108
    constrain v108 == u1 0 call f1([u8 69, u8 120, u8 2⁴×7, u8 101, u8 99, u8 116, u8 101, u8 100, u8 2⁵, u8 120, u8 2⁵, u8 33, u8 61, u8 2⁵, u8 121, u8 44, u8 2⁵, u8 98, u8 117, u8 116, u8 2⁵, u8 103, u8 111, u8 116, u8 2⁵, u8 98, u8 111, u8 116, u8 104, u8 2⁵, u8 101, u8 113, u8 117, u8 97, u8 108, u8 2⁵, u8 123, u8 120, u8 125], Field 1, v0, v109)
    constrain v108 == u1 0
    v110 = add v0, v1
    v111 = eq v110, v1
    v112 = not v111
    constrain v111 == u1 0 call f1([u8 69, u8 120, u8 2⁴×7, u8 101, u8 99, u8 116, u8 101, u8 100, u8 2⁵, u8 122, u8 2⁵, u8 33, u8 61, u8 2⁵, u8 121, u8 44, u8 2⁵, u8 98, u8 117, u8 116, u8 2⁵, u8 103, u8 111, u8 116, u8 2⁵, u8 98, u8 111, u8 116, u8 104, u8 2⁵, u8 101, u8 113, u8 117, u8 97, u8 108, u8 2⁵, u8 123, u8 122, u8 125], Field 1, v110, v112)
    constrain v0 == v1 call f2([u8 69, u8 120, u8 2⁴×7, u8 101, u8 99, u8 116, u8 101, u8 100, u8 2⁵, u8 120, u8 2⁵, u8 61, u8 61, u8 2⁵, u8 121, u8 44, u8 2⁵, u8 98, u8 117, u8 116, u8 2⁵, u8 120, u8 2⁵, u8 105, u8 115, u8 2⁵, u8 123, u8 120, u8 125, u8 2⁵, u8 97, u8 110, u8 100, u8 2⁵, u8 121, u8 2⁵, u8 105, u8 115, u8 2⁵, u8 123, u8 121, u8 125], Field 2, v0, v1, v108)
    return 
}
``` 
It honestly isn't horrible (especially for smaller messages), but perhaps we should add a flag to hide the value list of a call for constrain errors?